packages/paste-core/components/pagination/src/PaginationItems.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import {useUIDSeed} from '@twilio-paste/uid-library';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import type {PaginationItemsProps} from './types';
+import {PaginationItemsPropTypes} from './proptypes';
+
+const PaginationItems = React.forwardRef<HTMLUListElement, PaginationItemsProps>(({children, ...props}, ref) => {
+  const [childrenCount, validChildren] = React.useMemo(
+    () => [
+      React.Children.count(children),
+      React.Children.toArray(children).filter((child) => React.isValidElement(child) || typeof child === 'string'),
+    ],
+    [children]
+  );
+  const keySeed = useUIDSeed();
+
+  return (
+    <Box {...safelySpreadBoxProps(props)} ref={ref} as="ul" display="inline-flex" margin="space0" padding="space0">
+      {validChildren.map((child, index) => {
+        return (
+          <Box
+            key={keySeed(`pagination-items-${index}`)}
+            as="li"
+            listStyleType="none"
+            margin="space0"
+            marginRight={childrenCount !== index + 1 ? 'space90' : 'space0'}
```

Comment:
is this some kind of first child, last child conditional styles?

Reply:
Yeah. Same thing we're doing with Stack. I'm mapping through the children, wrapping them in an `li` with a `marginRight`, then setting the `marginRight` of the last `li` to `space0`.

Reply:
Does `_firstChild` `_lastChild` not work? Did we implement it? Are there too many wrappers?